src/shadowbox/server/manager_service.ts

```diff
@@ -224,22 +229,22 @@ export class ShadowsocksManagerService {
   }
 
   public getShareMetrics(req: RequestType, res: ResponseType, next: restify.Next): void {
-    res.send(200, {metricsEnabled: this.metricsPublisher.isSharingEnabled()});
+    res.send(Success.OK, {metricsEnabled: this.metricsPublisher.isSharingEnabled()});
     next();
   }
 
   public setShareMetrics(req: RequestType, res: ResponseType, next: restify.Next): void {
-    const params = req.params as SetShareMetricsParams;
-    if (typeof params.metricsEnabled === 'boolean') {
-      if (params.metricsEnabled) {
-        this.metricsPublisher.startSharing();
-      } else {
-        this.metricsPublisher.stopSharing();
-      }
-      res.send(204);
+    if (!req.params || typeof req.params.metricsEnabled !== 'boolean') {
+      next(new restify.BadRequestError(
+          `Expected metricsEnabled to be boolean.  Instead got ${req.params.metricsEnabled}`));
```

Comment:
The interpolated value may throw if `req` is undefined.  I suggest not echoing it and saying something like  'Must provide a boolean value for metricsEnabled'.

Reply:
I split it into two error cases so we can keep the richer error message.